salt/netapi/rest_cherrypy/app.py

```diff
@@ -1422,6 +1443,26 @@ def POST(self, **kwargs):
         else:
             creds = cherrypy.serving.request.lowstate
 
+        # Check acl for the api.
+        failure_str = ("[api_acl] Authentication failed for "
+                       "user {0} from IP {1}")
+        success_str = ("[api_acl] Authentication sucessful for "
+                       "user {0} from IP {1}")
+        user = creds.get('username', '')
+
+        # Check if handling a request from a proxy.
+        ip = cherrypy.request.headers.get('X-Forwarded-For', None)
```

Comment:
Using the X-Forwarded-For header has a couple problems:

- First off, its not a single ip-- its the chain of proxies and the client for this request (https://en.wikipedia.org/wiki/X-Forwarded-For)
- Second off, this header can't really be trusted... since its just a header (meaning anyone can set it to whatever they want). 

I'd suggest not supporting proxies, unless its needed as it can be a bit difficult on the API side to determine what the actual client is. If support is needed, then it should be configurable (on/off-- and ideally to let the user pick a header-- since its common for people's setups to have a specific header which *is* sanitized).